### ISSUE TITLE
`input-time-picker` + `focus-trap-disabled` should close when keyboard focus moves away 

### ISSUE THREAD

**Issue by jcfranco** [Esri/calcite-design-system#10639]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Actual Behavior

When tabbing outside of an open, focus-trap-disabled, `input-time-picker`'s input or time-picker parts, the component stays open.

### Expected Behavior

When focus moves outside of the component, it should close. This will be consistent with other components (e.g., `input-date-picker`, `combobox`)

### Reproduction Sample

https://codepen.io/jcfranco/pen/dyxdrax?editors=100

### Reproduction Steps

1. Tab to `input-time-picker`
2. Press down
3. Tab until focus moves out
4. Notice component stays open

### Reproduction Version

2.13.2

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

Calcite (dev)

**Comment by geospatialem** [comment]
Related to the consistency aimed with https://github.com/Esri/calcite-design-system/issues/10522

**Comment by driskull** [comment]
Ideally, this would use aria-activedescendant and not focus within the popover at all.

Only the input would be focusable and the arrow keys would allow for navigating between Hours/Minutes/Seconds/AMPM and arrow up/down would control increasing them. The user wouldn't be able to type in the input. Then a user wouldn't need to tab through the component at all.

**Comment by jcfranco** [comment]
Yeah. We could make this change once https://github.com/Esri/calcite-design-system/pull/11712 lands to reinforce that the picker is exclusively for pointer interaction. cc @eriklharper @ashetland

**Comment by jcfranco** [comment]
This is no longer reproducible in 3.1.0 (see https://codepen.io/jcfranco/pen/ZYYvzEe?editors=100). Closing!
